modules/global-classes/global-classes-rest-api.php

```
@@ -117,109 +99,77 @@ private function all() {
-			] )
-			->all();
-
-		$class = $this->get_repository()->get( $id );
-
-		if ( null === $class ) {
-			return Error_Builder::make( 'entity_not_found' )
-				->set_message( __( 'Global class not found', 'elementor' ) )
-				->set_status( 404 )
-				->build();
+		if ( $error ) {
+			return $error;
 		}
 
-		[$is_valid, $parsed, $errors] = Style_Parser::make( Style_Schema::get() )->parse( $value );
+		$order = $request->get_param( 'order' );
 
-		if ( ! $is_valid ) {
-			return Error_Builder::make( 'invalid_data' )
+		if ( ! $this->is_valid_order( $order, $sanitized_items ) ) {
+			return Error_Builder::make( 'invalid_order' )
 				->set_status( 400 )
-				->set_message( join( ', ', $errors ) )
+				->set_message( 'Invalid order' )
 				->build();
 		}
 
-		$updated = $this->get_repository()->put( $id, $parsed );
-		$order = $this->get_repository()->all()->get_order()->all();
+		$this->get_repository()->put(
+			$sanitized_items,
+			$order
+		);
 
-		return Response_Builder::make( (object) $updated )
-			->set_meta( [ 'order' => $order ] )
-			->build();
+		return Response_Builder::make()->no_content()->build();
 	}
 
-	private function post( \WP_REST_Request $request ) {
-		$class = Collection::make( $request->get_params() )
-			->only( [ 'label' ] )
-			->merge( [
-				'variants' => $request->get_param( 'variants' ) ?? [],
-				'type' => 'class',
-			] )
-			->all();
-
-		[ $is_valid, $parsed, $errors ] = Style_Parser::make( Style_Schema::get() )
-			->without_id()
-			->parse( $class );
-
-		if ( ! $is_valid ) {
-			return Error_Builder::make( 'invalid_data' )
-				->set_status( 400 )
-				->set_message( join( ', ', $errors ) )
-				->build();
+	private function sanitize_items( array $items ) {
+		$sanitized_items = [];
+
+		foreach ( $items as $item_id => $item ) {
+			[$is_valid, $sanitized, $errors] = Style_Parser::make( Style_Schema::get() )->parse( $item );
+
+			if ( ! $is_valid ) {
+				return [
+					null,
+					Error_Builder::make( 'invalid_item' )
```

Wouldn't it be better to throw an error and, in the put method, return the response?

---

Done